site/tutorials/keptn-notification-service.md

```diff
@@ -0,0 +1,277 @@
+
+Create a new `Dockerfile` with the following content:
+```
+FROM php:apache
+
+COPY index.php /var/www/html/index.php
+RUN chown -R www-data:www-data /var/www
+
+CMD ["apache2-foreground"]
+```
+
+## Create index.php
+Duration: 00:02:00
+
+In the same folder as your `Dockerfile` create another file called `index.php`:
+```
+<?php
+
+/* Create and open the log file
+ * Root directory is /var/www/html/
+ * So this log file is: /var/www/html/demoService.log
+ */
+$logFile = fopen("demoService.log", "a") or die("Unable to open file!");
+
+// Get the incoming data from Keptn
+$entityBody = file_get_contents('php://input');
+
+// Decode the incoming JSON event
+$cloudEvent = json_decode($entityBody);
+
+/* Your logic here...
+ * Write a log file entry into the /var/www/html/demoService.log
+ */
+fwrite($logFile,"Project Created: " . $cloudEvent->{'data'}->{'project'} . "\n\n");
+
+// Close handle to log file
+fclose($logFile);
+?>
+```
+
+Whenever the distributor detects an `sh.keptn.internal.event.project.create` event, it passes the event to this custom service.
+
+This service will parse the incoming keptn event then write a log line. This log file is stored at `/var/www/html/demoService.log` inside the `demo-service` pod.
+
+The log line will look like this:
+```
+Project Created: <project-name>
+```
+
+## Build and Push Image
+Duration: 00:02:00
+
+Build and push your custom service to DockerHub. Be sure to substitute your Docker ID into the following command:
+
+```
+docker build -t <your-docker-username>/keptn-demo-service:notification . && docker push <your-docker-username>/keptn-demo-service:notification
+```
+
+## Deploy Your Keptn Service
+Duration: 00:05:00
+
+Create a file on the Keptn machine called `demo-service.yaml`.
+
+Be sure to substitute **your** docker ID into the `image:` line
+
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: demo-service
+  namespace: keptn
+spec:
+  selector:
+    matchLabels:
+      run: demo-service
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: demo-service
+    spec:
+      containers:
+      - name: demo-service
+        image: <your-docker-username>/keptn-demo-service:notification
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 80
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: demo-service
+  namespace: keptn
+  labels:
+    run: demo-service
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: demo-service
+```
+
+Apply this file:
+```
+kubectl apply -f demo-service.yaml
+```
+
+Verify that both the `service` and `distributor` pods are running:
+
+```
+kubectl get pods -n keptn | grep demo
+
+demo-service-*-*                1/1     Running   0   24s
+demo-service-distributor-*-*    1/1     Running   0   39s
+```
+
+## Create Shipyard file
```

Comment:
probably rename it to "Create your Keptn project" or "Test your service" so that the users knows that the service implementation is finished at this point and now the actual testing starts.

Reply:
you could also move this part to the next section that is already called "Create Keptn project"